pkg/windows/windows.go

```diff
@@ -417,50 +408,44 @@ func (vm *windows) EnsureRequiredServicesStopped() error {
 	return nil
 }
 
-// ensureServicesAreRemoved ensures that all services installed by WMCO are removed from the instance
-func (vm *windows) ensureServicesAreRemoved() error {
-	// TODO: In this transitional period of migrating services to WICD's responsibility, WICD must be stopped first to
-	//       ensure it does not start any of these stopped services. This code should be removed as part of
-	//       https://issues.redhat.com/browse/WINC-733
-	svc := &service{name: wicdServiceName}
+// ensureServiceIsRemoved ensures that the given service installed by WMCO is removed from the instance
+func (vm *windows) ensureServiceIsRemoved(svc *service) error {
+	// If the service is not installed, do nothing
+	exists, err := vm.serviceExists(svc.name)
+	if err != nil {
+		return errors.Wrapf(err, "unable to check if %s service exists", svc.name)
+	}
+	if !exists {
+		return nil
+	}
+	// Make sure the service is stopped before we attempt to delete it
 	if err := vm.ensureServiceNotRunning(svc); err != nil {
-		return errors.Wrapf(err, "could not stop service %s", wicdServiceName)
+		return errors.Wrapf(err, "could not stop service %s", svc.name)
 	}
-	for _, svcName := range append(RequiredServices, AzureCloudNodeManagerServiceName) {
-		svc := &service{name: svcName}
-
-		// If the service is not installed, do nothing
-		exists, err := vm.serviceExists(svc.name)
-		if err != nil {
-			return errors.Wrapf(err, "unable to check if %s service exists", svc.name)
-		}
-		if !exists {
-			continue
-		}
-
-		// Make sure the service is stopped before we attempt to delete it
-		if err := vm.ensureServiceNotRunning(svc); err != nil {
-			return errors.Wrapf(err, "could not stop service %s", svc.name)
-		}
-		if err := vm.deleteService(svc); err != nil {
-			return errors.Wrapf(err, "could not delete service %s", svcName)
-		}
-		vm.log.Info("deconfigured", "service", svc.name)
+	if err := vm.deleteService(svc); err != nil {
+		return errors.Wrapf(err, "could not delete service %s", svc.name)
 	}
+	vm.log.Info("deconfigured", "service", svc.name)
 	return nil
 }
 
-func (vm *windows) Deconfigure() error {
+func (vm *windows) Deconfigure(preserveNode bool, apiServerURL string, credentials *Authentication) error {
```

Comment:
I don't see credentials being used here. Don't you need to get the wicdCAFile and wicdTokenFile values from it?

Reply:
I remoed the credentials param now. wicdCAFile and wicdTokenFile  are present in the k8sdir and their location is defined by constants, I am using them from there. Are you suggesting using ensureWICDSecretContent to verify their contents?

Reply:
No, I was just commenting on the unused parameter and wondering if you needed to do something with it.